### DISCUSSION TITLE
Use lock status as virtual key

### DISCUSSION THREAD

**Discussion by ghost** [houmain/keymapper#200] [Ideas]
I found that Fn+Esc, the Fnlock feature on lenovo thinkpads, can't be remapped because it doesn't send any key events besides the Fn key itself.  However I'm wondering if there's still some way for keymapper to to detect if the Fnlock is on or off. The same goes for other possible locks, though most of those can be remapped.

The use case for this would be, for example, to make the number keys act like function keys when the Fnlock is off.

**Comment by ristomatti** [comment]
I'm almost certain this is a firmware level feature and it's state is not exposed. The default state of the lock/mode can even be be set in BIOS am I right?

You could still do this though, just use some other key for it. For instance, toggle the mode by holding Esc for half a second, triggering a virtual modifier:

```ini
FnKey = F1 F2 F3 F4 F5 F6 F7 F8 F9 F10 F11 F12
FnLock = Virtual1

[modifier = "FnLock"]
  F1   >> # F1 alt function here 
  # ...
  F12  >> # F12 alt function here

[default]
  Escape{500ms} >> FnLock ^ FnLock
  FnKey         >> FnKey
```

What I'm doing myself though and suggest to consider. Map the function keys to their alternatives for all apps that either do not make use of function keys or in which you just never need/use them.

**Reply by ghost** [reply]
The thinkpad fnlock informs the OS about its status somehow, because it triggers an indicator on screen, same as Caps and Num.

**Reply by ristomatti** [reply]
I assume you're on Windows? I would've asked to check if you can see the key code on Linux, but I don't know if an equivalent tool exists for Windows.

I can test this at some point with my ThinkPad X1 Extreme 3rd Gen, but it's currently docked in a cumbersome enough place to not want to undock it now just to do this (it's basically a desktop PC for me as I work remote). 😄
